### ISSUE TITLE
Argument parser, tar error and working directory

### ISSUE THREAD

**Issue by aleximorin** [instructed-glacier-model/igm#7]
Hello,

I've managed to run a few simulations on my own windows laptop's cpu and on a linux cluster with a GPU. I've faced a few issues however and had to slightly modify the code however. It might be related to my poor comprehension of github and python packages, but here are the modifications I've had to make:

1. Using the argument parser didn't work. I've had to modify the ``common.py`` file so that the ``parser.parse_args(args=[])`` commands didn't pass an empty list but the arguments entered in the command line.

2. On windows, I've had to modify the ``tarfile.py`` file at line 2677 from ``name == member_name``  to ``name.replace(os.sep, '/') == member_name``. This is not related to ``IGM`` but rather ``OGGM``.  This code is called from using the ``OGGM_shop`` module. This line didn't properly manage different OS separators and windows used ``\\`` instead of ``/``. I didn't have to do this on the linux cluster.

3. Finally, using the OGGM shop preprocessing module downloaded some RGI files in the example folder defined by the working_dir argument. However, I was encountering an ``Access denied error`` at line 81 in the ``oggm_shop.py`` file. Trying to open the ``gridded_data.nc`` file raised this error as the path to the folder is not defined. Copy/pasting the RGI folder in the IGM folder where the ``igm_run.py`` file is located fixed the problem. I encountered this problem on both windows and linux. I suspect that this is linked to my poor installation from github but I'm not sure.

Thank you for this great tool! It was easy enough to generate results in little time.

**Comment by jouvetg** [comment]
Hi Alexi, 
Thanks a lot for your feedback on IGM, this is very useful!
1. did you encounter the argument parser issue on windows?
2. Indeed, OGGM is not recommended for windows (https://docs.oggm.org/en/stable/installing-oggm.html), and we are aware of the path issue. I'm surprised you solved this issue (this is great) only by changing one line of code, however, the file tarfile.py is not part on OGGM, right? I may ask OGGM developers for a fix if that's one line, but if I understand, your fix is not in OGGM code?
3. Ok, thanks I will investigate this.
Cheers
Guillaume

**Comment by aleximorin** [comment]
No worries Guillaume, it's a pleasure! I've had the chance today to dwelve a bit more into the code (mostly the preprocessing). I'm no software engineer but it seems really well built!

1. I encountered it on both Linux and Windows. I think it's just caused by passing the empty list ``args=[]`` to the argument parser. Maybe this was set to ease debugging? Removing it fixes the issue.

3. I've had to modify the ``tarfile.py`` file which seems to be related to the ``tarfile`` library. It's only dealing with proper os separators. Funnily enough, they're dealing with this issue on some other lines within the same file but not this one. 

4. I've re-read my issue and this point is not so clear. I've looked into it a bit and it seems an easy enough fix. 
For example, in the ``oggm_shop.py`` file, modifying line 81 from
``nc = Dataset(os.path.join(params.oggm_RGI_ID, "gridded_data.nc"), "r+")``
to
``nc = Dataset(os.path.join(params.working_dir, params.oggm_RGI_ID, "gridded_data.nc"), "r+")``
would fix it. However, some other modules are also calling trying to open files within this path ``os.path.join(params.oggm_RGI_ID, "some_file")``. Maybe you could either modify every module to take this into account, or simply modify the ``params.oggm_RGI_ID`` parameter to directly consider the full path.

**Comment by jouvetg** [comment]
all clear!
1. I understand, you used the working_dir parameter, which I introduced but never used nor tested, therefore I understand now why it does not work, I need to have a deeper look as I remember there a good reason to include args=[]
2. OK, understood, I will provide the fix to windows users. 
3. thanks, I fixed it, and committed, next time, don't hesitate to make a pull request to get credited of your contribution
thanks again!
